src/liquid/components/ld-notification/test/ld-notification.spec.ts

```diff
@@ -0,0 +1,772 @@
+        '.ld-notification__item:not(.ld-notification__item--dismissed)'
+      )
+      expect(notifications.length).toEqual(1)
+      expect(notifications[0].textContent).toContain('info #1')
+    })
+
+    it('does not do anything on ldNotificationDismiss event if no notifications are queued', async () => {
+      const page = await newSpecPage({
+        components: [LdNotification],
+        html: `<ld-notification></ld-notification>`,
+      })
+      await page.waitForChanges()
+
+      page.win.dispatchEvent(new CustomEvent('ldNotificationDismiss'))
+      await page.waitForChanges()
+
+      expect(page.root).toEqualHtml(`
+        <ld-notification
+          aria-label="Notifications"
+          class="ld-notification ld-notification--top"
+          role="region">
+        </ld-notification>
+      `)
+    })
+
+    it('dismisses all notifications on ldNotificationClear event', async () => {
+      const page = await newSpecPage({
+        components: [LdNotification],
+        html: `<ld-notification></ld-notification>`,
+      })
+      page.win.dispatchEvent(
+        new CustomEvent('ldNotificationAdd', {
+          detail: {
+            content: 'info #1',
+            type: 'info',
+          },
+        })
+      )
+      await page.waitForChanges()
+
+      page.win.dispatchEvent(
+        new CustomEvent('ldNotificationAdd', {
+          detail: {
+            content: 'info #2',
+            type: 'info',
+          },
+        })
+      )
+      await page.waitForChanges()
+
+      let notifications = page.root.querySelectorAll(
+        '.ld-notification__item:not(.ld-notification__item--dismissed)'
+      )
+      expect(notifications.length).toEqual(2)
+
+      page.win.dispatchEvent(new CustomEvent('ldNotificationClear'))
+      await page.waitForChanges()
+
+      notifications = page.root.querySelectorAll(
+        '.ld-notification__item:not(.ld-notification__item--dismissed)'
+      )
+      expect(notifications.length).toEqual(0)
+    })
+  })
+
+  describe('transitions', () => {
+    it('transitions single notifications', async () => {
+      jest.useFakeTimers()
+
+      const page = await newSpecPage({
+        components: [LdNotification],
+        html: `<ld-notification></ld-notification>`,
+      })
+      page.win.dispatchEvent(
+        new CustomEvent('ldNotificationAdd', {
+          detail: {
+            content: 'info #1',
+            type: 'info',
+          },
+        })
+      )
+      await page.waitForChanges()
+
+      page.win.dispatchEvent(
+        new CustomEvent('ldNotificationAdd', {
+          detail: {
+            content: 'info #2',
+            type: 'info',
+          },
+        })
+      )
+      await page.waitForChanges()
+
+      page.win.dispatchEvent(
+        new CustomEvent('ldNotificationAdd', {
+          detail: {
+            content: 'info #3',
+            type: 'info',
+          },
+        })
+      )
+      await page.waitForChanges()
```

Comment:
I'd assume that the waitForChanges call is only necessary after the last notification was added?

Reply:
True, I have removed the calls in multiple places. I might have missed some, but they should hurt anybody.